### DISCUSSION TITLE
Horizontal TreeTab? Or just a Tab layout?

### DISCUSSION THREAD

**Discussion by jrwrigh** [qtile/qtile#4484] [Ideas]
I'm wanting to get something akin to i3's tabbed layout, which looks like this:
![image](https://github.com/qtile/qtile/assets/20801821/00a49ee4-de88-4d71-a47e-8925f8576a34)

In essence, it's a `TreeTab` layout without sections and with the tabs horizontal rather than vertical. I see two possible ways of going about doing that:

## 1) Add a horizontal option to the `TreeTab` layout
I'm not sure how this would work with the actual "tree" part of the `TreeTab` layout. Having the tree displayed vertically is very natural, while I have no idea how it'd look horizontal.

I could have it error out if there's an attempt to add a section to a `TreeTab` layout with horizontal tabs, but that seems messy/unintuitive

## 2) Add a new `Tab` layout with vertical/horizontal options

This seems like the most logical options. Basically just strip away all the "tree" parts of `TreeTab`.

### Feedback
First, what are thoughts between the two options?

Second, which, if any options, would be accepted into `qtile`? I know the project seems happy with it's current layouts and wants to push new layouts as separate projects, but this seems simple enough that it perhaps warrants inclusion in the main project.

**Comment by elParaguayo** [comment]
Just to comment on your last paragraph: that's not true, new layouts are absolutely welcome and we're happy to include them where it makes sense to do so.

**Reply by jrwrigh** [reply]
Ah, ok. I must've misremembered a previous discussion (which would've probably been from at least a year ago, so it makes sense).

**Reply by elParaguayo** [reply]
Yes, there was a discussion about the ScreenSplit layout PR. We discussed whether it should go into qtile-extras but it will go into the main repo once I've tidied it up.

**Comment by m-col** [comment]
The regular tab layout seems like an attractive option that I would imagine people would be interested in using, particularly people who like it from i3/sway, whereas a horizontal treetab might be a bit more niche? But ultimately whichever one you want :)

**Reply by jrwrigh** [reply]
Cool. I'll see if I can carve out some time to make a quick prototype. Don't imagine it being too difficult, but who knows.

**Comment by hanschen** [comment]
I have a proof-of-concept for a maximized Tabbed layout with horizontal tabs (vertical would be nice, I just haven't found time to implement it yet). See the discussion here: https://github.com/qtile/qtile/issues/491#issuecomment-1741795387

Feel free to extend this concept if you find it useful!

I would love to have tabs for the stacked mode in the `Columns` layout as well. This would be a big selling point for people considering Qtile, in my opinion (I personally hesitated to move from i3 due to the lack of tabs).
